src/unix/linux-core.c

```diff
@@ -811,29 +811,48 @@ uint64_t uv_get_constrained_memory(void) {
   uint64_t high;
   uint64_t max;
   char* p;
+  char* p_end;
 
   if (uv__slurp("/proc/self/cgroup", buf, sizeof(buf)))
     return uv__get_constrained_memory_fallback();
 
   /* In the case of cgroupv2, we'll only have a single entry. */
-  if (memcmp(buf, "0::/", 4))
-    return uv__get_constrained_memory_fallback();
+  if (0 == memcmp(buf, "0::/", 4)) {
+    p = strchr(buf, '\n');
+    if (p != NULL)
+      *p = '\0';
+
+    p = buf + 4;
+
+    snprintf(filename, sizeof(filename), "/sys/fs/cgroup/%s/memory.max", p);
+    max = uv__read_uint64(filename);
+
+    snprintf(filename, sizeof(filename), "/sys/fs/cgroup/%s/memory.high", p);
+    high = uv__read_uint64(filename);
+  } else {
+    p = strchr(buf, ':');
+    while (p != NULL && memcmp(p, ":memory:", 8)) {
```

Comment:
oh wait, I realise this isn't how cgroups work...

Reply:
Actually, I think it is correct? But it would be good for someone who understands these things to check.

Reply:
I take it slurm creates a cgroup subtree? On my cgroupv1 system, the limit sits at `/sys/fs/cgroup/memory` directly. It'd be useful to add a comment here about what you're doing, that's easier to review than to parse C code 🙂

Reply:
Yes, that appears to be how cgroupv1 works, e.g. https://unix.stackexchange.com/a/621576

Reply:
Apparently we're still missing something though, as on my cgroup1 system with Docker, I can't access even the memory subgroup listed in `/proc/self/cgroup`.